dgc/DgCertificate.cs

```diff
@@ -1,23 +1,45 @@
-namespace DGC
-{
-    using System;
-    using System.Collections.Generic;
-
-    using System.Globalization;
-    using Newtonsoft.Json;
-    using Newtonsoft.Json.Converters;
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
 
+namespace DCC
+{
     /// <summary>
-    /// EU Digital Green Certificate
+    /// EU Digital Green Certificate v.1.3.0
     /// </summary>
     public partial class DgCertificate
     {
         /// <summary>
-        /// Date of Birth of the person addressed in the DGC. ISO 8601 date format restricted to
-        /// range 1900-2099
+        /// Date of birth of the DCC holder.  
+        /// Complete or partial date without time restricted to the range from 1900-01-01 to 2099-12-31.  
+        /// Exactly 1 (one) non-empty field MUST be provided if the complete or partial date of birth is known.If the date of birth is not known even partially, the field MUST be set to an empty string "". This should match the information as provided on travel documents.
+        /// One of the following ISO 8601 formats MUST be used if information on date of birth is available.Other options are not supported.
+        /// YYYY-MM-DD
+        /// YYYY-MM
+        /// YYYY
         /// </summary>
         [JsonProperty("dob")]
-        public DateTimeOffset DateOfBirth { get; set; }
+        public string DateOfBirthString { get; set; }
+
+        [JsonIgnore]
+        public DateTime? DateOfBirth 
+        { 
+            get {
+                if (DateTime.TryParse(DateOfBirthString, out var res))
+                {
+                    return res;
+                } 
+                else if (int.TryParse(DateOfBirthString, out int year))
+                {
+                    return new DateTime(year, 1, 1);
```

Comment:
Why 1,1 args here?

Reply:
so it will be a valid date. so I dob is "2000" then the dateofbirth field will be 2000.1.1.
This may differ between countries but dob is always a date in Iceland and if the day is not known then it is the first of that month.